src/main/nav_create_simulated_image.py

```
@@ -946,11 +1046,18 @@ def _rebuild_dynamic_tabs(self) -> None:
+                    elif widget_kind == 'ring' and widget_data_index is not None:
+                        if 0 <= widget_data_index < len(self.sim_params['rings']):
+                            ring_name = self.sim_params['rings'][widget_data_index].get(
+                                'name', f'Ring{widget_data_index+1}')
+                            target_tab_name = ring_name
                     elif widget_kind == 'star' and widget_data_index is not None:
                         if 0 <= widget_data_index < len(self.sim_params['stars']):
                             star_name = self.sim_params['stars'][widget_data_index].get(
                                 'name', f'Star{widget_data_index+1}')
                             target_tab_name = star_name
+                    else:
+                        assert False, f'Unknown kind: {widget_kind}'
```

Ruff (0.14.11)

Assertion always fails, replace with `pytest.fail()`

(PT015)